"Fixes" small pockets, Nerfs hat storage, Limits candleboxes, Removes sign storage

VickiMorris opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Modifies small pockets (/datum/component/storage/concrete/pockets/small) to only allow storing small items instead of normal items, bringing it in line with how tiny pockets work for storage. Makes it so the candlepack can only store candles, lighters, and matchboxes instead of _any_ five tiny or small sized items and ~~are now small sized themselves~~ cannot be stored on belts anymore. Removed the exo storage component from the janitor's wet floor signs.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

As it stands, the fedora and taqiyahs (which are a type of skullcap, mind you) allow storing a medium/normal sized item, which includes such comical events of a detective pulling; A sawn-off shotgun, a skateboard, an entire mop, the cap's jetpack, a batong, a wooden buckler, a _silenced_ Stechkin, or a laptop out from under their hat, where you were none the wiser to it being there. This is a frankly insanely strong level of smuggling, as most people don't even check boots or cloaks, let alone hats.

Candleboxes can basically be used at bootleg omnibelts/gear harnesses right now. It's a 5 slot belt item that can hold any small or tiny item without any restrictions, it's frankly too strong IMO. ~~This also changes them to be small-sized, to make up for the lost utility and now that 'storage-maxxing' isn't possible with them anymore.~~

While it's funny for wet floor signs to grant exo storage as they are a type of suit, it doesn't exactly make sense (it's two flaps of plastic) and can currently lead to some disgusting storage maxxing as they're also small items. You could nest 7 in a box and sudden double its storage to 14, or one in each of your pockets to have 4 small items instead of just two.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![Clothing Storage Tweaks](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/0cb6b5cf-54bf-4e5e-a334-feac137615ed)

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/d10c40fd-0922-479a-a171-10743e17921b

</details>

## Changelog
:cl: Impish_Delights
balance: Fedoras (Regular and All Det subtypes) and Taqiyahs only store upto small items (down from normal-sized).
balance: Candle packs now only hold candles, lighters, and matchboxes, and cannot be stored on belts.
balance: Wet Floor Signs no longer have exo storage like other suit items do.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

VickiMorris committed:
Small Pockets, Limited Candleboxes, No Sign Storage

Small pockets (used in fedora and taqiyahs) can now only hold small items, similar to tiny pockets with tiny items.
Candleboxes can only hold candles, lighters, and matchboxes (no longer any 5 small or tiny items)
Wet floor signs no longer have exo storage

VickiMorris committed:
Smaller candle packs

Candle packs are now small instead of medium (normal) sized

Rukofamicom commented on code/game/objects/items/storage/fancy.dm:
```diff
@@ -109,6 +109,7 @@
 	item_state = "candlebox5"
+	w_class = WEIGHT_CLASS_SMALL
 	throwforce = 2
 	slot_flags = ITEM_SLOT_BELT
```
```suggestion
```

I recommend simply removing the ability to store in the belt slot instead of changing the other properties of the box. 

This both makes more sense and is more consistent with other boxes that are almost entirely normal sized.

VickiMorris replied:
Adding storage type limitations brings the candlebox in line with every other type of "fancy" box (cigar cases, cigarette packs, nugget boxes, egg cartons). Keeping it as a belt-slot lets the Chaplain have (although only slightly) quicker access to candles on their person even if they ditch their backpack (as some nullrod transforms are back-slot only and too large to be stored in a backpack). This basically turns it into a cigarette pack but for candles instead.

Rukofamicom replied:
I more meant with the fact it makes very little sense to have a box that fits in your belt slot. 

Similarly didn't like First-aid boxes in belts, but it was done for mechanical reasons anyway (and is about to be reversed)

VickiMorris replied:
I'll axe the belt slot and pop it back to medium/normal-sized when I get home. I'm gonna keep the item restrictions listing though as it still does the "generates more space" thing otherwise (5 small items in a medium box)

Rukofamicom replied:
That is pretty much what I was trying to suggest for it!

VickiMorris replied:
Hehe brain being bad at interperating things strikes again

VickiMorris committed:
Reembiggens & Debelts Candleboxes

Candleboxes no longer can be stored on the belt slot.
Candleboxes are normal sized once again.
